swarm/api/http/tmpl/htmlstrtmpl.go

```diff
@@ -0,0 +1,375 @@
+// Copyright 2017 The go-ethereum Authors
+// This file is part of the go-ethereum library.
+//
+// The go-ethereum library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The go-ethereum library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the go-ethereum library. If not, see <http://www.gnu.org/licenses/>.
+
+/*
+We use html templates to handle simple but as informative as possible error pages.
+
+To eliminate circular dependency in case of an error, we don't store error pages on swarm.
+We can't save the error pages as html files on disk, or when deploying compiled binaries 
+they won't be found.
+
+For this reason we resort to save the HTML error pages as strings, which then can be
+parsed by Go's html/template package
+*/
+package tmpl
```

Comment:
I suggest just putting them in a separate file (e.g. `html.go`) inside the `swarm/api/http` package.